examples/rpc/Server.cpp

```diff
@@ -0,0 +1,65 @@
+#include "generated/echo/Console.pb.hpp"
+#include "hal/generic/TimerServiceGeneric.hpp"
+#include "protobuf/echo/Echo.hpp"
+#include "services/network_instantiations/NetworkAdapter.hpp"
+#include <iostream>
+
+class Console
+    : public examples::Console
+{
+public:
+    Console(services::Echo& echo)
+        : examples::Console(echo)
+    {}
+
+    void Write(infra::BoundedConstString message) override
+    {
+        std::cout << infra::AsStdString(message) << std::endl;
+        MethodDone();
+    }
+};
+
+class EchoConnection
+    : public services::EchoOnConnection
+{
+public:
+    EchoConnection()
+        : console(*this)
+    {}
+
+private:
+    Console console;
+};
+
+class RpcServer
+    : public services::ServerConnectionObserverFactory
+{
+public:
+    RpcServer(services::ConnectionFactory& factory)
+        : listener(factory.Listen(1234, *this))
+    {}
+
+    void ConnectionAccepted(infra::AutoResetFunction<void(infra::SharedPtr<services::ConnectionObserver> connectionObserver)>&& createdObserver, services::IPAddress address) override
+    {
+        if (connection)
+            connection->::services::EchoOnConnection::Subject().AbortAndDestroy();
+
+        if (connection.Allocatable())
+            createdObserver(connection.Emplace());
+    }
+
+private:
+    infra::SharedPtr<void> listener;
+    infra::SharedOptional<EchoConnection> connection;
+};
+
+int main(int argc, const char* argv[], const char* env[])
```

Comment:
TIL that 'env' is a thing.

From cppreference:
> A very common implementation-defined form of main() has a third argument (in addition to argc and argv), of type char**, pointing at [an array of pointers to the execution environment variables](https://pubs.opengroup.org/onlinepubs/9699919799/functions/exec.html).

Never heard of this, _ever_, before.

Reply:
It's a bit of legacy in our examples. We used to have an environment variable pointing to a configuration file. We used the 'env' argument to find it. Now it's not used like that anymore but the 'env' in the main signature stayed.